articles/search/cognitive-search-skill-pii-detection.md

```diff
@@ -101,25 +99,38 @@ Parameters are case-sensitive and all are optional.
 
 ```json
 {
-  "values": [
-    {
-      "recordId": "1",
-      "data" : 
-      {
-        "piiEntities":[ 
-           { 
-              "text":"859-98-0987",
-              "type":"U.S. Social Security Number (SSN)",
-              "subtype":"",
-              "offset":28,
-              "length":11,
-              "score":0.65
-           }
-        ],
-        "maskedText": "Microsoft employee with ssn *********** is using our awesome API's."
-      }
-    }
-  ]
+    "documents": [
```

Comment:
These changes do not line up with the output that the Azure Cognitive Search skill produces. In our API, we have different names of something because the skill was originally written to work with a preview version of the language API (specifically it was v3.0-preview.1). Unfortunately as the language folks changed things, we had to map the changed names (i.e. category -> type, confidenceScore->score, redactedText -> maskedText once language added their own support for redaction, etc) because it would be a breaking change for our customers to switch to the new naming. It is unfortunate that we were unable to keep the APIs in sync because of this.

TL;DR the proposed changes here are actually incorrect to the Azure Cognitive Search API due to legacy backwards compatibility and should not be merged.

Reply:
@Careyjmac do you mind elaborating a little more on that? Maybe I was using the wrong Azure Cognitive Search API / endpoint and mixed the inputs/outputs from the documentation.

Thank you!

Reply:
Do you have an Azure Cognitive Search service, and are you creating a skillset and adding the 'Sample definition' from this doc to it? Some helpful docs for this:
https://docs.microsoft.com/en-us/azure/search/cognitive-search-defining-skillset
https://docs.microsoft.com/en-us/azure/search/cognitive-search-tutorial-blob

The 'Sample Output' schema is then the schema that you should be using in your index definition, similar to this:
   `` {
      "name": "pii_entities",
      "type": "Collection(Edm.ComplexType)",
      "analyzer": null,
      "synonymMaps": [],
      "fields": [
        {
          "name": "text",
          "type": "Edm.String",
          "facetable": false,
          "filterable": false,
          "key": false,
          "retrievable": true,
          "searchable": true,
          "sortable": false,
          "analyzer": "standard.lucene",
          "indexAnalyzer": null,
          "searchAnalyzer": null,
          "synonymMaps": [],
          "fields": []
        },
        {
          "name": "type",
          "type": "Edm.String",
          "facetable": false,
          "filterable": false,
          "key": false,
          "retrievable": true,
          "searchable": true,
          "sortable": false,
          "analyzer": "standard.lucene",
          "indexAnalyzer": null,
          "searchAnalyzer": null,
          "synonymMaps": [],
          "fields": []
        },
        {
          "name": "subtype",
          "type": "Edm.String",
          "facetable": false,
          "filterable": false,
          "key": false,
          "retrievable": true,
          "searchable": true,
          "sortable": false,
          "analyzer": "standard.lucene",
          "indexAnalyzer": null,
          "searchAnalyzer": null,
          "synonymMaps": [],
          "fields": []
        },
        {
          "name": "offset",
          "type": "Edm.Int32",
          "facetable": false,
          "filterable": false,
          "retrievable": true,
          "sortable": false,
          "analyzer": null,
          "indexAnalyzer": null,
          "searchAnalyzer": null,
          "synonymMaps": [],
          "fields": []
        },
        {
          "name": "length",
          "type": "Edm.Int32",
          "facetable": false,
          "filterable": false,
          "retrievable": true,
          "sortable": false,
          "analyzer": null,
          "indexAnalyzer": null,
          "searchAnalyzer": null,
          "synonymMaps": [],
          "fields": []
        },
        {
          "name": "score",
          "type": "Edm.Double",
          "facetable": false,
          "filterable": false,
          "retrievable": true,
          "sortable": false,
          "analyzer": null,
          "indexAnalyzer": null,
          "searchAnalyzer": null,
          "synonymMaps": [],
          "fields": []
        }
      ]
    },
    {
      "name": "masked_text",
      "type": "Edm.String",
      "facetable": false,
      "filterable": false,
      "key": false,
      "retrievable": true,
      "searchable": true,
      "sortable": false,
      "analyzer": "standard.lucene",
      "indexAnalyzer": null,
      "searchAnalyzer": null,
      "synonymMaps": [],
      "fields": []
    }``

And you will need output field mappings in your indexer definition similar to this:

 `` "outputFieldMappings": [
    {
      "sourceFieldName": "/document/content/pii_entities",
      "targetFieldName": "pii_entities"
    },
    {
      "sourceFieldName": "/document/content/masked_text",
      "targetFieldName": "masked_text"
    }
  ],``

Alternatively you can also use the [Import data wizard](https://docs.microsoft.com/en-us/azure/search/search-import-data-portal) in the portal and  under 'Add cognitive skills (Optional)'->'Add Enrichments' select 'Extract personally identifiable information' and it will do it all for you automatically so you don't have to worry about the JSON definitions.